HPC/WRF-v422-patch-correctness.diff

```diff
@@ -0,0 +1,119 @@
+diff --git a/arch/configure.defaults b/arch/configure.defaults
+index 692fbfe9..5a465e91 100644
+--- a/arch/configure.defaults
++++ b/arch/configure.defaults
+@@ -775,6 +775,8 @@ LD              =       $(FC)
+ RWORDSIZE       =       CONFIGURE_RWORDSIZE
+ PROMOTION       =       #-fdefault-real-8
+ ARCH_LOCAL      =       -DNONSTANDARD_SYSTEM_SUBR  -DWRF_USE_CLM
++# to validate WRF correctness on aarch64 against x86_64; compile with x86 this alternate ARCH_LOCAL option; able to achieve identical output model on ubuntu 18.04 with gcc 10.2 (on graviton and intel processors)
++ARCH_LOCAL      =       -DNONSTANDARD_SYSTEM_SUBR  -DWRF_USE_CLM -DAARCH64_X86_CORRECTNESS_FIX
+ CFLAGS_LOCAL    =       -w -O3 -c  # -DRSL0_ONLY
+ LDFLAGS_LOCAL   =       
+ CPLUSPLUSLIB    =       
+@@ -1982,6 +1984,49 @@ LIB_BUNDLED     = \
+ 
+ #insert new stanza here
+ 
++###########################################################
++#ARCH    Linux aarch64, GCC compiler OpenMPI # serial smpar dmpar dm+sm
++#
++DESCRIPTION     =      GCC ($SFC/$SCC): Aarch64
++DMPARALLEL      =
++OMPCPP          =      -fopenmp
++OMP             =      -fopenmp
++OMPCC           =      -fopenmp
++SFC             =      gfortran
++SCC             =      gcc
++CCOMP           =      gcc
++DM_FC           =      mpif90
++DM_CC           =      mpicc -DMPI2_SUPPORT
++FC              =      CONFIGURE_FC
++CC              =      CONFIGURE_CC
++LD              =      $(FC)
++RWORDSIZE       =      CONFIGURE_RWORDSIZE
++PROMOTION       =
++ARCH_LOCAL      =      -DAARCH64_X86_CORRECTNESS_FIX
++CFLAGS_LOCAL    =      -w -O3 -c
++LDFLAGS_LOCAL   =      -fopenmp
++FCOPTIM         =      -Ofast -mcpu=native -funroll-loops -fno-expensive-optimizations -fno-reciprocal-math -fsigned-zeros -fno-unsafe-math-optimizations
++# for graviton 2 use the folowing flag
++#FCOPTIM         =      -Ofast -march=armv8.2-a+fp16+rcpc+dotprod -funroll-loops -fno-expensive-optimizations -fno-reciprocal-math -fsigned-zeros -fno-unsafe-math-optimizations
```

Comment:
why not -march=native here?

Reply:
You should probably put that in a comment.